homeassistant/components/hyperion/config_flow.py

```diff
@@ -0,0 +1,379 @@
+            await self.async_set_unique_id(hyperion_id)
+            self._abort_if_unique_id_configured()
+        else:
+            return self.async_abort(reason="no_id")
+        return await self.async_step_auth()
+
+    async def async_step_user(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        if not user_input:
+            return self._show_setup_form()
+        self._data = user_input
+        return await self.async_step_auth()
+
+    async def _cancel_request_token_task(self) -> None:
+        """Cancel the request token task if it exists."""
+        if self._request_token_task is not None:
+            if not self._request_token_task.done():
+                self._request_token_task.cancel()
+
+            # Cancellation is only processed on the **next** cycle of the event loop, so
+            # yield here.
+            await asyncio.sleep(0)
+
+            try:
+                await self._request_token_task
+            except asyncio.CancelledError:
+                pass
+            self._request_token_task = None
+
+    async def _request_token_task_func(self, auth_id: str) -> Optional[ConfigType]:
+        """Send an async_request_token request."""
+        auth_resp = {}
+        async with await self._create_client(raw_connection=True) as hyperion_client:
+            if hyperion_client:
+                # The Hyperion-py client has a default timeout of 3 minutes on this request.
+                auth_resp = await hyperion_client.async_request_token(
+                    comment=DEFAULT_ORIGIN, id=auth_id
+                )
+            await self.hass.config_entries.flow.async_configure(
+                flow_id=self.flow_id, user_input=auth_resp
+            )
+        return auth_resp
+
+    def _get_hyperion_url(self):
+        """Return the URL of the Hyperion UI."""
+        # If this flow was kicked off by SSDP, this will be the correct frontend URL. If
+        # this is a manual flow instantiation, then it will be a best guess (as this
+        # flow does not have that information available to it). This is only used for
+        # approving new dynamically created tokens, so the complexity of asking the user
+        # manually for this information is likely not worth it (when it would only be
+        # used to open a URL, that the user already knows the address of).
+        return f"http://{self._data[CONF_HOST]}:{self._port_ui}"
+
+    async def async_step_auth(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle the auth step of a flow."""
+
+        # First connect without attempting to login and determine if
+        # authentication is required.
+        async with await self._create_client(raw_connection=True) as hyperion_client:
+            if not hyperion_client:
+                return self._show_setup_form({CONF_BASE: "connection_error"})
```

Comment:
We can't reuse the form for the user step in other steps. The form is step specific and we should only move between step functions by await a step function, ie not go directly to another steps' form.